app/repositories/onlinetesting/Phase2TestRepository.scala

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories.onlinetesting
+
+import factories.DateTimeFactory
+import model.ApplicationStatus.ApplicationStatus
+import model.Exceptions.UnexpectedException
+import model.OnlineTestCommands.OnlineTestApplication
+import org.joda.time.DateTime
+import model.persisted.{ CubiksTest, Phase2TestProfile }
+import model.persisted.{ ExpiringOnlineTest, NotificationExpiringOnlineTest, Phase2TestProfileWithAppId, TestResult }
+import model.ProgressStatuses.{ PHASE1_TESTS_INVITED, _ }
+import model.{ ApplicationStatus, ProgressStatuses, ReminderNotice }
+import play.api.Logger
+import reactivemongo.api.DB
+import reactivemongo.bson._
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait Phase2TestRepository extends OnlineTestRepository[CubiksTest, Phase2TestProfile] {
+  this: ReactiveRepository[_, _] =>
+
+  def getTestGroup(applicationId: String): Future[Option[Phase2TestProfile]]
+
+  def getTestProfileByToken(token: String): Future[Phase2TestProfile]
+
+  def getTestProfileByCubiksId(cubiksUserId: Int): Future[Phase2TestProfileWithAppId]
+
+  def insertOrUpdateTestGroup(applicationId: String, phase2TestProfile: Phase2TestProfile): Future[Unit]
+
+  def nextTestGroupWithReportReady: Future[Option[Phase2TestProfileWithAppId]]
+
+  def updateGroupExpiryTime(applicationId: String, expirationDate: DateTime): Future[Unit]
+
+  def removeTestProfileProgresses(appId: String, progressStatuses: List[ProgressStatus]): Future[Unit]
+
+  def insertTestResult(appId: String, phase2Test: CubiksTest, testResult: TestResult): Future[Unit]
+
+  def nextTestForReminder(reminder: ReminderNotice): Future[Option[NotificationExpiringOnlineTest]]
+
+  def nextExpiringApplication: Future[Option[ExpiringOnlineTest]]
+}
+
+class Phase2TestMongoRepository(dateTime: DateTimeFactory)(implicit mongo: () => DB)
+  extends ReactiveRepository[Phase2TestProfile, BSONObjectID]("application", mongo,
+    model.persisted.Phase2TestProfile.phase2TestProfileFormat, ReactiveMongoFormats.objectIdFormats
+  ) with Phase2TestRepository {
+
+  val phaseName = "PHASE2"
```

Comment:
I think we should def look at doing that, maybe after the e-tray release goes out?  Or at the weekend if I'm feeling fastidious.